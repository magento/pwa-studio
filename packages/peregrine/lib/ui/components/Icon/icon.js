import React from 'react';
import { object, number, shape, string } from 'prop-types';

import { useStyle } from '../../classify';
import defaultClasses from './icon.module.css';

const Icon = props => {
    // destructure `propClasses` to exclude it from `restProps`
    const {
        attrs,
        classes: propClasses,
        size,
        src: Component,
        ...restProps
    } = props;
    const { width, ...restAttrs } = attrs || {};
    const classes = useStyle(defaultClasses, propClasses);

    return (
        <span className={classes.root} {...restProps}>
            <Component
                className={classes.icon}
                size={size || width}
                {...restAttrs}
            />
        </span>
    );
};

export default Icon;

Icon.propTypes = {
    attrs: shape({}),
    classes: shape({
        icon: string,
        root: string
    }),
    size: number,
    src: object.isRequired
};
