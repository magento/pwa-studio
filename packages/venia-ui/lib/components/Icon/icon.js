import React from 'react';
import { func, number, shape, string } from 'prop-types';

import { useStyle } from '../../classify';
import defaultClasses from './icon.css';

const Icon = props => {
    // destructure `propClasses` to exclude it from `restProps`
    const {
        attrs,
        classes: propClasses,
        size,
        src: Component,
        ...restProps
    } = props;
    const { width, ...restAttrs } = attrs || {};
    const classes = useStyle(defaultClasses, propClasses);

    return (
        <span className={classes.root} {...restProps}>
            <Component
                className={classes.icon}
                size={size || width}
                {...restAttrs}
            />
        </span>
    );
};

export default Icon;

Icon.propTypes = {
    attrs: shape({}),
    classes: shape({
        icon: string,
        root: string
    }),
    size: number,
    src: func.isRequired
};
