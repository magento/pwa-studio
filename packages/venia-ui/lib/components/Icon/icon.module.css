.root {
    align-items: center;
    display: inline-flex;
    justify-content: center;
    touch-action: manipulation;
}

.icon {
    fill: var(--fill, none);
    stroke: var(--stroke, rgb(var(--venia-global-color-text)));
}

@media (min-width: 641px) {
    .icon_desktop {
        width: 28px;
        height: 28px;
    }
}
