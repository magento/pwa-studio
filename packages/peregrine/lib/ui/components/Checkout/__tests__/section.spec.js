import React from 'react';
import testRenderer from 'react-test-renderer';
import Section from '../section';
import Icon from '../../Icon';

test('renders a Section component', () => {
    const component = testRenderer.create(<Section />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('renders with icon if showEditIcon is true', () => {
    const component = testRenderer.create(<Section showEditIcon={true} />);
    expect(() => component.root.findByType(Icon)).not.toThrow();
});

test('renders without icon if showEditIcon is false', () => {
    const component = testRenderer.create(<Section showEditIcon={false} />);
    expect(() => component.root.findByType(Icon)).toThrow();
});
