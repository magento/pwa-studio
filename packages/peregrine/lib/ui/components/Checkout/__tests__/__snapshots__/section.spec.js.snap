// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Section component 1`] = `
<button>
  <span>
    <span>
      <span />
    </span>
    <span />
    <span />
  </span>
</button>
`;
